test/integration/cli-domain-package-server-script/cli-domain-package-server-script.js

```diff
@@ -386,5 +387,77 @@ describe('cli : domain : package-server-script', function(){
         );
       });
     });
+
+    describe('when component uses es2015', function(){
+      describe('and requires a local js module', function(){
+        describe('and requires a json file', function(){
+          describe('and includes for loops', function(){
+            var someJsonContent = JSON.stringify({firstName: 'John', lastName: 'Doe'});
+            var someJsModuleContent = 'const infiniteLoops = () => {' +
+              'let x, y, z;' +
+              'while(true){ x = 234; }' +
+              'for(var i=1e12;;){ y = 546; }' +
+              'do { z = 342; } while(true);' +
+              '};' +
+              'const sayHello = (name) => `Hello ${name}`;' +
+              'export default sayHello;';
+
+            var serverContent = 'import sayHello from \'./someModule\';' +
+              'import user from \'./someData\';' +
+              'const {firstName: first, lastName: last} = user;' +
+              'const hello = sayHello(`${first} ${last}`);' +
+              'const data = (context, callback) => callback(null, { hello });' +
+              'export {data}';
+
+            beforeEach(function(done){
+              fs.writeFileSync(path.resolve(componentPath, 'someModule.js'), someJsModuleContent);
+              fs.writeFileSync(path.resolve(componentPath, 'someData.json'), someJsonContent);
+              fs.writeFileSync(path.resolve(componentPath, serverName), serverContent);
+              done();
+            });
+
+            afterEach(function(done){
+              require.cache[path.resolve(publishPath, serverName)] = null;
+              done();
+            });
+
+            it('should save compiled data provider', function(done){
+              packageServerScript(
```

Comment:
what if we bring all of this stuff inside the beforeEach and we save the result (compiledContent) into a variable, so then we can have 3 individual it sync tests that can explain a little bit better what we are doing? 
e.g. first test we have no error, then test wrapping correctly works for do, for, and while? I think it would improve readability and maintainability